plugins/callback/foreman.py

```diff
@@ -106,30 +161,29 @@
 from ansible.plugins.callback import CallbackBase
 
 
-def build_log(data):
+def build_log_legacy(data):
     """
     Transform the internal log structure to one accepted by Foreman's
     config_report API.
     """
-    for source, msg in data:
-        if msg.get('failed'):
-            level = 'err'
-        elif msg.get('changed'):
-            level = 'notice'
-        else:
-            level = 'info'
-
-        yield {
-            "log": {
-                'sources': {
-                    'source': source,
-                },
-                'messages': {
-                    'message': json.dumps(msg, sort_keys=True),
-                },
-                'level': level,
-            }
+    if data.get('failed'):
+        level = 'err'
+    elif data.get('changed'):
+        level = 'notice'
+    else:
+        level = 'info'
+
+    yield {
+        "log": {
+            'sources': {
+                'source': data.get('module'),
```

Comment:
Previously this was a task name (e.g. `Create /tmp/file file`), but with this change it will be the module name (e.g. `file`). Since we don't want to change existing behaviour, maybe we shouldn't change this?

Reply:
Good find, I have created "name" key that will hold the previous value. And I am changing this to "name" as well.